src/logger.ts

```diff
@@ -0,0 +1,352 @@
+    this.tenantId = process.env.RAD_SECURITY_TENANT_ID || "";
+  }
+
+  configure(config: Partial<LoggerConfig>): void {
+    if (config.minLevel !== undefined) {
+      this.minLevel = config.minLevel;
+    }
+    if (config.enableStderr !== undefined) {
+      this.enableStderr = config.enableStderr;
+    }
+    if (config.enableMcpNotifications !== undefined) {
+      this.enableMcpNotifications = config.enableMcpNotifications;
+    }
+    if (config.stderrFormat !== undefined) {
+      this.stderrFormat = config.stderrFormat;
+    }
+  }
+
+  setServer(server: Server, sessionId?: string): void {
+    this.server = server;
+    this.sessionId = sessionId;
+  }
+
+  setLevel(level: LogLevel): void {
+    this.minLevel = level;
+  }
+
+  getLevel(): LogLevel {
+    return this.minLevel;
+  }
+
+  private shouldLog(level: LogLevel): boolean {
+    return LOG_LEVEL_PRIORITY[level] >= LOG_LEVEL_PRIORITY[this.minLevel];
+  }
+
+  private sanitizeData(data: any): any {
+    if (typeof data !== 'object' || data === null) {
+      return data;
+    }
+
+    const sanitized = Array.isArray(data) ? [...data] : { ...data };
+
+    const sensitiveKeys = [
+      'token',
+      'sessionToken',
+      'session_token',
+      'accessKeyId',
+      'access_key_id',
+      'secretKey',
+      'secret_key',
+      'password',
+      'authorization',
+      'cookie',
+      'api_key',
+      'apiKey',
+    ];
+
+    for (const key in sanitized) {
+      const lowerKey = key.toLowerCase();
+      if (sensitiveKeys.some(sk => lowerKey.includes(sk.toLowerCase()))) {
+        sanitized[key] = '[REDACTED]';
+      } else if (typeof sanitized[key] === 'object' && sanitized[key] !== null) {
+        sanitized[key] = this.sanitizeData(sanitized[key]);
+      }
+    }
+
+    return sanitized;
+  }
+
+  private async sendLog(level: LogLevel, logger: string, data: any): Promise<void> {
+    if (!this.server) {
+      return;
+    }
+
+    try {
+      const enrichedData = {
+        timestamp: new Date().toISOString(),
+        ...data,
+      };
+
+      if (this.accountId) {
+        enrichedData.account_id = this.accountId;
+      }
+      if (this.tenantId) {
+        enrichedData.tenant_id = this.tenantId;
+      }
+
+      const params: any = {
+        level,
+        logger,
+        data: enrichedData
+      };
+
+      if (this.sessionId) {
+        params._meta = {
+          sessionId: this.sessionId
+        };
+      }
+
+      await this.server.sendLoggingMessage(params);
+    } catch (error) {
+      // If notification fails (e.g., not connected yet), log to stderr
+      // The original log already happened, so info isn't lost
+      if (error instanceof Error) {
+        if (error.message !== "Not connected") {
+          console.error(`[LOGGER ERROR] Failed to send MCP notification: ${error.message}`);
+        }
+      }
+    }
+  }
+
+  private formatTimestamp(): string {
+    const now = new Date();
```

Comment:
why not RFC3339 for JSON?

Reply:
the [toISOString()](https://github.com/rad-security/mcp-server/blob/95b1ce40771369cea9076b31ac1b1845862789d7/src/logger.ts#L206) outputs ISO 8601 format